tests/test_utils.sh

```diff
@@ -710,6 +710,31 @@ function test_is_function {
   unalias not_my_function
 }
 
+function test_hash_color {
+    PS1="$ "
+    lp_activate --no-config # For having _lp_foreground
+    _lp_hash_color "Debug"
+    assertContains "$lp_hash_color" "Debug"
+    # FIXME How to test color?
+}
+
+function test_join {
+    _lp_join "_" "A" "B " " " " C" " D " "EE"
+    assertEquals "$lp_join" "A_B _ _ C_ D _EE"
+
+    _lp_join "-" "A"
+    assertEquals "$lp_join" "A"
+
+    _lp_join "+" ""
+    assertEquals "$lp_join" ""
+
+    typeset -a arr
+    arr=(1 2 3)
+    _lp_join "/" "${arr[@]}"
+    assertEquals "$lp_join" "1/2/3"
+}
+
```

Comment:
You have the expected and actual arguments switched around in all of these tests, it should be `assertEquals "1/2/3" "$lp_join"`

Reply:
Out of curiosity, why does it matter? I would understand for `assertContains`, but for `assertEquals`?

Reply:
Because the output of the test error labels each string incorrectly:

> ASSERT:expected:<A_B   C D EE> but was:<A_B _ _ C_ D _EE>